### PR TITLE
add modules aliases for absolute imports

### PR DESCRIPTION
**Pull request by reactoholic**
### Update all relative imports in the client application

Note:

With this PR some additional (unplanned) changes are getting in:
- types instead of interfaces;
- FC usage reduced;
- other syntaxis logic applied by add-on;

There should be minimal to none logical changes.

### CODE REVIEW

**Comment by reactoholic** [comment]
@coderabbitai pause

**Comment by hero101** [comment]
https://medium.com/@devxprite/how-to-setup-path-aliases-in-vite-df955939ffe8

I also recommend building a production version to test if this is reliable.

**Comment by reactoholic** [comment]
> https://medium.com/@devxprite/how-to-setup-path-aliases-in-vite-df955939ffe8
> 
> I also recommend building a production version to test if this is reliable.

I did the updates recomended in the article and build the app locally - no errors were thrown.

File: src/core/auth/authentication/pages/AuthRequiredPage.tsx [hunk 1]
```diff
@@ -28,23 +28,25 @@ export const AuthRequiredPage: FC<AuthRequiredPageProps> = () => {
...
-      <Box display="flex" marginTop={4} gap={2} justifyContent="center">
-        <Button component={Link} to={loginUrl} variant="outlined" color="primary">
-          {t('authentication.sign-in')}
-        </Button>
-        <Button component={Link} to={signUpUrl} variant="outlined">
-          {t('authentication.sign-up')}
-        </Button>
-      </Box>
-    </AuthenticationLayout>
+    <>
```
Comment (ccanos) [inline src/core/auth/authentication/pages/AuthRequiredPage.tsx:31]: please revert this one

File: src/core/ui/card/CardFooter.tsx [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-      alignItems="center"
-      height={gutters(2)}
-      paddingX={1}
-      {...containerProps}
-    >
-      {children}
-    </Box>
-  );
-};
+const CardFooter = ({ children, ...containerProps }: PropsWithChildren<BoxProps>) => (
```
Comment (ccanos) [inline src/core/ui/card/CardFooter.tsx:5]: I think BoxProps already has children. but leave it like this, should be fine

**Comment by reactoholic** [comment]
Svetlio has pushed my branch into `dev` and I tested the routes - every route worked, no errors were thrown from any component what so ever.

**Review by bobbykolev — APPROVED** [review]
The volume of this PR is enormous. I'm struggling with just leaving a comment and approving it.
Building it locally and even tested on sandbox should be sufficient.

However, there were some side changes that are hard to track. 
Let's merge and in case of any regressions, we'll quickly act. This should be a one-time effort and occasion where we're touching side things inside a PR with such size.

File: src/core/ui/forms/FormikAutocomplete.tsx [hunk 3]
```diff
@@ -28,23 +28,20 @@ export const FormikAutocomplete: FC<FormikAutocompleteProps> = ({
...
 
-  const helperText = useMemo(() => {
-    if (!isError) {
-      return helpText;
-    }
-
-    return tErr(meta.error as TranslationKey, { field: name });
-  }, [isError, meta.error, helpText, name, tErr]);
+  const helperText = useMemo(
+    () => (isError ? tErr(meta.error as TranslationKey, { field: name }) : helpText),
```
Comment (ccanos) [inline src/core/ui/forms/FormikAutocomplete.tsx:39]: please revert this

**Review by ccanos — APPROVED** [review]

**Pull request merged**